src/test/java/org/apache/commons/dbcp2/TestPoolableConnection.java

```diff
@@ -187,4 +191,22 @@ public void testPoolableConnectionLeak() throws Exception {
 
         assertEquals(0, pool.getNumActive(), "The pool should have no active connections");
     }
+
+    @Test
+    public void testIsDisconnectionSqlExceptionStackOverflow() throws Exception {
+        int maxDeep = 70000;
+        SQLException rootException = new SQLException("Data truncated", "22001");
+        SQLException parentException = rootException;
+        for (int i = 0; i <= maxDeep; i++) {
+            SQLException childException = new SQLException("Data truncated:" + i, "22001");
+            parentException.setNextException(childException);
+            parentException = childException;
+        }
+        final Connection conn = pool.borrowObject();
+        Method method = conn.getClass().getDeclaredMethod("isDisconnectionSqlException", SQLException.class);
```

Comment:
If you make the method package private, you should not need reflection.

Reply:
Did you mean to make `isDisconnectionSqlException()` package-private?

Reply:
The method you are trying to call.

Reply:
I have made the amendment as you suggested.